### DISCUSSION TITLE
Smart Meter Interface as an additional input device

### DISCUSSION THREAD

**Discussion by nobodyman1** [arendst/Tasmota#15955] [Q&A] [answered]
I want to use the Smart Meter Interface as an additional input device.
I plan to use the SONOFF for measuring the delivered energy from a photovoltaic system (Balkonkraftwerk, Mini-PV-Anlage). Therefore I think I place the SONOFF device near to my Smart Meter and connect it via an infrared reader to the SONOFF. Then I´m able to get delivered and consumed energy with one device!
I think is is very useful!
It only needs a simple infrared header connected to the SONOFF board. There is a header with 3.3V, Rx, Tx and Gnd available.

**Comment by arendst** [comment, marked as answer]
We do not support ANY sensor on an AC connected to Gnd device. All POWs have this connection so you may find 230V on your Gnd connection.

I suggest you change the header of this discussion by removing any reference to POW (or any other energy measuring device).

**Reply by nobodyman1** [reply]
Ok, I didn't notice the AC connection!

**Comment by nobodyman1** [comment]
But what´s the difference to the SONOFF TH Elite (THR316D) which has external sensors?

**Reply by barbudor** [reply]
On every devices that is performing power measurement, the ESP is not isolated from the mains.
ThR is isolated

**Reply by nobodyman1** [reply]
Ok, I understand. I looked into some schematic diagrams. It's no good idea to add an external device.
